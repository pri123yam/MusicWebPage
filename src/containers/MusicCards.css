.Card{
    height: auto;
    width: 22%;
    display: inline-block;
    border: solid;
    margin: 0.10%;
    position: relative;
    border-color: black;
    border-radius: 18px;
}
.CardImage{
    border-radius: 18px;
    height: 35vh;
    width:100%;
}