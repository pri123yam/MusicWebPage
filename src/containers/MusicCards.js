import React, { Component } from 'react';
import { connect } from 'react-redux';
import {
    Card, CardImg, CardText, CardBody,
    CardTitle, CardSubtitle
} from 'reactstrap';
import './MusicCards.css';
import DisplaySongs from '../selectors/DisplaySongSelector';

export class MusicCards extends Component {
    render() {
        return this.props.songs.map((item) => {
            return (
                <div key={item.id} className='Card'>
                    <Card>
                        <CardImg className='CardImage'  src={item.thumbnail} alt="Card image cap" />
                        <CardBody>
                            <CardTitle>name:{item.name}</CardTitle>
                            <CardSubtitle>genre:{item.genre}</CardSubtitle>
                            <CardText>artist:{item.artist}</CardText>
                        </CardBody>
                    </Card>
                </div>
            );
        });
    }
}
function mapStateToProps(state) {
    return {
        songs: DisplaySongs(state)
    };
}

export default connect(mapStateToProps)(MusicCards);
