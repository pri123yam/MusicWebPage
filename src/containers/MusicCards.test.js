import React from 'react';
import {MusicCards} from './MusicCards';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({
    adapter: new Adapter()
});

describe("<MusicCards> Component",()=>{
    let initProps;
    let wrapper;
    beforeEach(()=>{
        initProps={
            songs:[
                    {
                        id:1,
                        name:"enchanted",
                        genre:"pop",
                        artist:"taylor swift",
                        duration:"5:53",
                        thumbnail:''
                    },
                    {
                        id:2,
                        genre: "pop",
                        name:"what makes you beautiful",
                        artist:"one direction",
                        duration:"3:18",
                        thumbnail:""
                    }
                ]}
    });

    it("should render <MusicCards> component correctly",()=>{
        wrapper=Enzyme.shallow(<MusicCards
                                    songs={initProps.songs}
                                    />);
        expect(wrapper.debug()).toMatchSnapshot();
    })
})