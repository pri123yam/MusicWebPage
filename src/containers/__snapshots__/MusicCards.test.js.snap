// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MusicCards> Component should render <MusicCards> component correctly 1`] = `
"<div className=\\"Card\\">
  <Card tag=\\"div\\">
    <CardImg className=\\"CardImage\\" src=\\"\\" alt=\\"Card image cap\\" tag=\\"img\\" />
    <CardBody tag=\\"div\\">
      <CardTitle tag=\\"div\\">
        name:
        enchanted
      </CardTitle>
      <CardSubtitle tag=\\"div\\">
        genre:
        pop
      </CardSubtitle>
      <CardText tag=\\"p\\">
        artist:
        taylor swift
      </CardText>
    </CardBody>
  </Card>
</div>


<div className=\\"Card\\">
  <Card tag=\\"div\\">
    <CardImg className=\\"CardImage\\" src=\\"\\" alt=\\"Card image cap\\" tag=\\"img\\" />
    <CardBody tag=\\"div\\">
      <CardTitle tag=\\"div\\">
        name:
        what makes you beautiful
      </CardTitle>
      <CardSubtitle tag=\\"div\\">
        genre:
        pop
      </CardSubtitle>
      <CardText tag=\\"p\\">
        artist:
        one direction
      </CardText>
    </CardBody>
  </Card>
</div>"
`;
